### PR TITLE
Dwarf EH: add test cases

### PR DESCRIPTION
**Pull request by WalterBright**
It would have saved me many hours of debugging if these had already been in the test suite.

### CODE REVIEW

File: test/runnable/eh.d [hunk 1]
```diff
@@ -755,6 +755,73 @@ void test12989()
...
+    throw new Exception("msg");
+    goto L3;
+}
+
+void test10()
+{
+    int x;
+    try
+    {
+	bar10(0x110000);
```
Comment (andralex) [inline test/runnable/eh.d]: misaligned

File: test/runnable/eh.d [hunk 2]
```diff
@@ -755,6 +755,73 @@ void test12989()
...
+    }
+    catch (Exception e)
+    {
+	printf("caught\n");
```
Comment (andralex) [inline test/runnable/eh.d]: bunch of misaligned lines?

File: test/runnable/eh.d [hunk 3]
```diff
@@ -755,6 +755,73 @@ void test12989()
 
 /****************************************************/
 
+int bar10(int c)
+{
+    if (c <= 0xFFFF)
+    {
+    L3:
```
Comment (andralex) [inline test/runnable/eh.d]: Is the position of this label important? Otherwise you could put it together with the goto: `L3: goto L3;`

**Comment by andralex** [comment]
Looks like there's a spaces/tabs/indent issue with this code. Please fix?

**Comment by WalterBright** [comment]
tabs fixed

File: test/runnable/eh.d (hunk 3 above)
Comment (WalterBright) [inline test/runnable/eh.d]: Yes, it has to be there.

**Comment by andralex** [comment]
thx!

**Comment by WalterBright** [comment]
Well, that was quick! Tanks!

**Comment by MartinNowak** [comment]
Can you please inform us about the state of the new EH?
http://forum.dlang.org/post/56806778.2040204@dawg.eu
We also need a [changelog](https://github.com/D-Programming-Language/dmd/blob/dbba59cadafe540c0c98d69ac867173335e145cd/changelog.dd) entry describing how, why, and on which platform we changed the EH.

From a recent comment on yours I inferred that there is no performance improvement (are we still spilling all registers, is the finally block still a [call thrashing the return prediction](https://issues.dlang.org/show_bug.cgi?id=12442#c3)).
In any case we should run a few microbenchmarks to ensure that we're not introducing a performance regression.

**Comment by WalterBright** [comment]
It works on my system, but the more thorough autotester has discovered some more issues with it. The first version will be Linux 64 only. It should be the same speed in the no EH case, and slower when throwing. I have not taken advantage of optimization opportunities with it, as the focus is getting it to work.

**Comment by MartinNowak** [comment]
Thanks for the update. The beta will start on 1st Jan running until 14th Jan, meaning we'll merge master into stable in 3 days. Remember every fix after the merge needs to target stable.
Do you think 2 weeks to sort out the final issues will suffice?

**Pull request merged**